src/twisted/internet/test/test_tcp.py

```diff
@@ -115,6 +115,28 @@ def connect(client, destination):
 
 
 
+class TCPClientTestsBuilder(ReactorBuilder):
+    """
+    Builder defining tests relating to L{IReactorTCP.connectTCP}.
+    """
+    def _freePort(self, interface='127.0.0.1'):
+        probe = socket.socket()
```

Comment:
This is a *bad* way of probing for a free port; it's non-deterministic and race-condition-prone .